src/Arcus.Security.Providers.CommandLine/CommandLineSecretProvider.cs

```diff
@@ -62,8 +56,6 @@ public Task<string> GetRawSecretAsync(string secretName)
         /// <exception cref="SecretNotFoundException">Thrown when the secret was not found, using the given name.</exception>
         public Secret GetSecret(string secretName)
         {
-            Guard.NotNullOrWhitespace(secretName, nameof(secretName), "Requires a non-blank secret name to look up the command line argument secret");
```

Comment:
we don't check for null here anymore because we rely on the null-check in `GetRawSecret` ?
Is that a good idea ?

Reply:
Normally, no, but in this case the percentage of overloading functionality is greater than actual functionality. Since we do not need to use the values, only pass it along. It makes it  bit more easier to follow + only validates when it needs it.

Think it is discussible of what would be best. In a team with lots of inputs/members, I would indeed do paranoia-checks; but I might be bit pragmatic in this case.

Eager to hear your thoughts on this.